presto-raptorx/src/main/java/com/facebook/presto/raptorx/util/IteratorPageSource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.util;
+
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.Page;
+
+import java.util.Iterator;
+
+import static java.util.Objects.requireNonNull;
+
+public class IteratorPageSource
```

Comment:
"highker on Feb 4  Member
This new util can be replace by existing FixedPageSource."

They 2 indeed look similar. FixedPageSource is last modified in 2015, and IteratorPageSource is created in 2017. I think they had a reason to create a new thing; otherwise they could just copy the same from V1.

This is only used in SystemTable, and only in V2. I'd rather put it in the above mentioned "refactor SystemTable" task.